tyssue/core/objects.py

```diff
@@ -191,12 +191,14 @@ def copy(self, deep_copy=True):
         """
         if deep_copy:
             datasets = {element: df.copy() for element, df in self.datasets.items()}
+            specs = {key: value.copy() for key, value in self.specs.items()}
```

Comment:
Ok, so this only works because `specs` is a dict of dict, so each `value` has a `copy` method, but this might not be true in general (i.e if a user does something like `specs['date'] = today()`).  It's more general to simply do
`specs = self.specs.copy()`

Reply:
I tried that first but it seems like it doesn't enable the deep copy of the dictionaries...
I could fix this with an if statement but I am not sure you'd appreciate !

Reply:
Ha, yes I recall I had that kind of problem. To avoid the if, you can use
```py
from copy import deepcopy
...
            specs = deepcopy(self.specs)
```

See:
https://docs.python.org/3/library/copy.html